### PR TITLE
issue #4514: Move CKernelNormalizer to BASE_CLASS

### PR DESCRIPTION
**Pull request by dmittov**

### CODE REVIEW

File: src/shogun/util/factory.h [hunk 1]
```diff
@@ -40,6 +41,7 @@ namespace shogun
 	CDistance* distance(const std::string& name);
 	CEvaluation* evaluation(const std::string& name);
 	CKernel* kernel(const std::string& name);
+    CKernelNormalizer* kernel_normalizer(const std::string& name);
```
Comment (vinx13) [inline src/shogun/util/factory.h]: style: use tab

File: examples/meta/src/binary/kernel_support_vector_machine.sg [hunk 2]
```diff
@@ -11,7 +11,9 @@ Labels labels_test = labels(f_labels_test)
 #![create_features]
 
 #![set_parameters]
+KernelNormalizer n = kernel_normalizer("IdentityKernelNormalizer")
```
Comment (karlnapf) [inline examples/meta/src/binary/kernel_support_vector_machine.sg]: maybe instead we could have a separate example for under "kernel" for the normalizer?

File: examples/meta/src/binary/kernel_support_vector_machine.sg [hunk 3]
```diff
@@ -11,7 +11,9 @@ Labels labels_test = labels(f_labels_test)
...
 Kernel k = kernel("GaussianKernel", log_width=1.0)
+k.set_normalizer(n)
```
Comment (karlnapf) [inline examples/meta/src/binary/kernel_support_vector_machine.sg]: this should be "put"

File: examples/meta/src/binary/kernel_support_vector_machine.sg (hunk 2 above)
Comment (dmittov) [reply examples/meta/src/binary/kernel_support_vector_machine.sg]: Check plz

File: src/interfaces/swig/Kernel.i [hunk 4]
```diff
@@ -61,12 +55,9 @@ PROTOCOLS_CUSTOMKERNEL(CustomKernel, float32_t, "f\0", NPY_FLOAT32)
 %rename(PolyMatchWordStringKernel) CPolyMatchWordStringKernel;
 %rename(SalzbergWordStringKernel) CSalzbergWordStringKernel;
 %rename(SimpleLocalityImprovedStringKernel) CSimpleLocalityImprovedStringKernel;
```
Comment (karlnapf) [inline src/interfaces/swig/Kernel.i]: the includes also need to be removed from `Kernel_includes.i`

**Comment by gf712** [comment]
i think this just need a rebase and could still be merged?

**Comment by karlnapf** [comment]
It'd be great to rebase/merge this!

**Pull request closed without merging**